examples/language-modeling/run_electra_pretraining.py

```diff
@@ -0,0 +1,752 @@
+
+            while len(features) > self.num_tensors_per_file:
+                feature_set_path = self.feature_directory / f"feature_set_{job_id}_{feature_index}.pt"
+                torch.save(features[: self.num_tensors_per_file], feature_set_path)
+                features = features[self.num_tensors_per_file :]
+                feature_index += 1
+                feature_set_paths.append(feature_set_path)
+
+        if len(features) > 0:
+            feature_set_path = self.feature_directory / f"feature_set_{job_id}_{feature_index}.pt"
+            torch.save(features, feature_set_path)
+            feature_set_paths.append(feature_set_path)
+
+        return feature_set_paths
+
+    @staticmethod
+    def parse_file(file_index):
+        try:
+            features = torch.load(file_index)
+            yield from features
+        except RuntimeError:
+            raise RuntimeError(f"Corrupted file {file_index}")
+
+    def __len__(self):
+        return len(self.feature_set_paths) * self.num_tensors_per_file
+
+    def __iter__(self):
+        return chain.from_iterable(map(self.parse_file, self.feature_set_paths))
+
+
+class CombinedModel(nn.Module):
+    def __init__(
+        self,
+        discriminator: PreTrainedModel,
+        generator: PreTrainedModel,
+        tokenizer: PreTrainedTokenizer,
+        training_args: ElectraTrainingArguments,
+        data_args: DataTrainingArguments,
+    ):
+        super().__init__()
+
+        self.discriminator = discriminator
+        self.generator = generator
+
+        # Embeddings are shared
+        self.discriminator.set_input_embeddings(self.generator.get_input_embeddings())
+
+        self.tokenizer = tokenizer
+
+        self.discriminator_weight = training_args.discriminator_weight
+        self.generator_weight = training_args.generator_weight
+        self.mask_probability = data_args.mask_probability
+        self.max_predictions_per_sequence = data_args.max_predictions_per_sequence
+
+        # Original implementation has a default of :(mask_probability + 0.005) * max_sequence_length
+        if self.max_predictions_per_sequence == -1:
+            self.max_predictions_per_sequence = (self.mask_probability + 0.005) * data_args.block_size
+
+        class Config:
+            xla_device: bool = False
+
+        self.config = Config()
+
+    def mask_inputs(
+        self, input_ids: torch.Tensor, tokens_to_ignore, proposal_distribution=1.0,
+    ):
+        input_ids = input_ids.clone()
+        inputs_which_can_be_masked = torch.ones_like(input_ids)
+        for token in tokens_to_ignore:
+            inputs_which_can_be_masked -= torch.eq(input_ids, token).long()
+
+        total_number_of_tokens = input_ids.shape[-1]
+
+        # Identify the number of tokens to be masked, which should be: 1 < num < max_predictions per seq.
+        # It is set to be: n_tokens * mask_probability, but is truncated if it goes beyond bounds.
+        number_of_tokens_to_be_masked = torch.max(
```

Comment:
Maybe simplify to avoid creating `tensor(1)`:

```python
torch.min(
    torch.tensor(self.max_predictions_per_sequence, dtype=torch.long),
    torch.tensor(int(total_number_of_tokens * self.mask_probability), dtype=torch.long),
).clamp(1)
```